### PR TITLE
[Release] Refactor User Tests

### PR DESCRIPTION
**Pull request by amogkam**
Refactors user tests so that they are easier to run in a separate repo

- Consolidates all the tests under one directory
- Moves all the driver setup into bash scripts instead of directly in `build_pipeline.py`
- Removes `ConnectTest` abstraction in favor of bash scripts that are specified in the test yaml file and run in `e2e.py`.
- Also reduces the number of workers for the rllib connect test

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ml_user_tests/tune_rllib/app_config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+base_image: "anyscale/ray-ml:pinned-nightly-py37-gpu"
```
Comment (gjoliver) [inline release/ml_user_tests/tune_rllib/app_config.yaml:1]: so I just fixed a minor issue where the long_running_tests were using a really dated app_config to run rllib stacks.
should we not create another copy of app_config for this, and simply point to: ../../rllib_tests/app_config.yaml ?

**Review by richardliaw — APPROVED** [review]
Tests passing? Looks good at a high level

File: release/ml_user_tests/tune_rllib/app_config.yaml (hunk 1 above)
Comment (amogkam) [reply release/ml_user_tests/tune_rllib/app_config.yaml:1]: Good point, I’ll do that in a follow up PR. Just going to merge this for now to unblock @alanwguo

**Comment by amogkam** [comment]
Yes, tests are passing!

**Pull request merged**